addon/components/inputs/when.js

```diff
@@ -167,22 +166,24 @@ export default AbstractInput.extend({
      * @returns {undefined}
      */
     selectedButton (event) {
+      const value = event.target.value
       const firstButtonValue = this.get('firstButtonValue')
 
-      // Set which button is selected.
-      // Needed because clicking in the date-time-picker changes the event object.
-      if (event.target.value === DATE_VALUE || event.target.value === firstButtonValue) {
-        this.set('selectedValue', event.target.value)
-      }
-
       // Set the bunsen model to the value of the selected button
       this.onChange(
         this.get('bunsenId'),
-        (event.target.value === firstButtonValue) ? firstButtonValue : this.get('storedDateTimeValue')
+        (value === firstButtonValue) ? firstButtonValue : (
```

Comment:
It's not exactly the same as previous implementation. I can try to make it clearer.